WalletWasabi/Tor/TorMonitor.cs

```diff
@@ -24,11 +26,12 @@ public class TorMonitor : PeriodicRunner
 {
 	public static readonly TimeSpan CheckIfRunningAfterTorMisbehavedFor = TimeSpan.FromSeconds(7);
 
-	public TorMonitor(TimeSpan period, Uri fallbackBackendUri, TorHttpClient httpClient, TorProcessManager torProcessManager) : base(period)
+	public TorMonitor(TimeSpan period, Uri fallbackBackendUri, TorProcessManager torProcessManager, HttpClientFactory backendHttpClientFactory) : base(period)
 	{
 		FallbackBackendUri = fallbackBackendUri;
-		HttpClient = httpClient;
 		TorProcessManager = torProcessManager;
+		TorHttpPool = backendHttpClientFactory.TorHttpPool!;
```

Comment:
Getting smellier.

Reply:
It can be null, so shouldn't it be handled like that?

Reply:
The call in line 120 can be problematic. await TorHttpPool**?**.ReportCircuitClosedAsync is OK IMO.

Reply:
It cannot be `null` because `TorMonitor` is instantiated only when Tor is enabled in user settings.

Reply:
But I'm not happy with L33 either. The thing is that if I refactor it more, the PR will get bigger and so far I don't know about a better way to do it. Any ideas are welcome.